src/Microsoft.DotNet.Darc/src/Darc/Operations/VirtualMonoRepo/VmrOperationBase.cs

```diff
@@ -117,9 +117,20 @@ SourceMapping ResolveMapping(string repo)
                 Logger.LogInformation("{message}", e.Message);
                 return true;
             }
+            catch (WorkBranchException e)
+            {
+                Logger.LogError("{message}", e.Message);
+                return false;
+            }
             catch (Exception e)
             {
-                Logger.LogError("Failed to synchronize repo {name}{exception}", mapping.Name, Environment.NewLine + e.Message);
+                Logger.LogError(
+                    "Failed to synchronize repo {name}{exception}." +
+                    Environment.NewLine +
+                    "A new branch was created for the sync and didn't get merged as the sync was interrupted. A new sync should start from the original branch.", 
```

Comment:
This only applies for recursive synchronization/initialization actually.

Also, if we run the logic from somewhere else (MSBuild task) than Darc, we also want to see this.

For these reasons, I think you should move this catch block inside the `VmrInitializer`/`VmrUpdater` where we deal with the recursive processes.

I'd imagine you can catch, log this extra line and throw the exception further.

Reply:
Also, if you do it in the right place, you can tell user which branch is the original one (the recursive methods know this when they create the work branch).

Reply:
so it has to be caught inside the `UpdateRepositoryRecursively` and the `_currentBranch` field needs to be public

Reply:
Possibly. Can it maybe be caught inside `MergeBack()`?

Reply:
there is a edge case - initializing a repo fails, current branch is `init/{mapping.Name}/..` then we run update, the original branch is the init branch, not main

Reply:
Maybe it's not possible, yeah